### DISCUSSION TITLE
[C128] Can't target C128 with linux compiler

### DISCUSSION THREAD

**Discussion by Divarin** [cc65/cc65#2439] [General]
Simple hello world program for C128:

<pre>
#include <stdio.h>
#include <c128.h>

void main()
{
	printf("Hello World!\n");
}
</pre>
Compile using Linux compiler:
`~/git/cc65/bin/cc65 -t c128 hello.c`
`~/git/cc65/bin/cl65 term.s -o term.prg`

Run in VICE (128), garbage on the screen.
Run in VICE (64) works fine.

Compile same source with windows compiler through wine:
`wine ~/work/code/cc65/bin/cc65.exe -t c128 term.c`
`wine ~/work/code/cc65/bin/cl65.exe term.s -o term.prg`

Run in VICE (128) works fine.
Run in VICE (64) nothing happens.

It looks like the Linux version of the compiler is not targeting C128 but actually C64.

**Comment by mrdudz** [comment]
You compiled for c128, but linked against the c64 libs :) you should give -t c128 to all tools, also to cl65

**Reply by polluks** [reply]
However Windows should have the same effect.

**Reply by Divarin** [reply]
okay adding -t c128 option worked. I didn't know about that as I didn't need to use it when I was compiling on windows.

**Reply by rofl0r** [reply]
maybe because you can't compare apples with oranges. your windows version is 4 years older than the linux one as you just admitted in the 2nd thread you opened.

**Comment by mrdudz** [comment]
Just coincidence really, it was always required
